File: POO JAVA/Aula07/src/aula07/Lutador.java
package aula07;

public class Lutador {
    
    //Atributos;
    private String nome;
    private String nacionalidade;
    private int idade;
    private float altura;
    private float peso;
    private String categoria;
    private int vitorias;
    private int derrotas;
    private int empate;
    
    //Metodos Construtores

    public Lutador(String nome, String nacionalidade, int idade, float altura, float peso, int vitorias, int derrotas, int empate) {
        this.nome = nome;
        this.nacionalidade = nacionalidade;
        this.idade = idade;
        this.altura = altura;
        this.setPeso(peso);
        this.setVitorias(vitorias);
        this.setDerrotas(derrotas);
        this.setEmpate(empate);
    }
    
    //Metodos Acessores e Modificadores;
    public String getNome() {
        return nome;
    }
    private void setNome(String nome) {
        this.nome = nome;
    }
    private String getNacionalidade() {
        return nacionalidade;
    }
    private void setNacionalidade(String nacionalidade) {
        this.nacionalidade = nacionalidade;
    }
    private int getIdade() {
        return idade;
    }
    private void setIdade(int idade) {
        this.idade = idade;
    }
    private float getAltura() {
        return altura;
    }
    private void setAltura(float altura) {
        this.altura = altura;
    }
    private float getPeso() {
        return peso;
    }
    private void setPeso(float peso) {
        this.peso = peso;
        setCategoria();
    }
    public String getCategoria() {
        return categoria;
    }
    private void setCategoria() {
        if(getPeso() < 52.2){
            this.categoria = "Invalido";
        }else if(getPeso() <= 70.3){
            this.categoria = "Leve";
        }else if(getPeso() <=83.9){
            this.categoria = "Medio";
        }else if(getPeso() <= 120.2){
            this.categoria = "Pesado";
        }else{
            this.categoria = "Invalido";
        }
    }
    private int getVitorias() {
        return vitorias;
    }
    private void setVitorias(int vitorias) {
        this.vitorias = vitorias;
    }
    private int getDerrotas() {
        return derrotas;
    }
    private void setDerrotas(int derrotas) {
        this.derrotas = derrotas;
    }
    private int getEmpate() {
        return empate;
    }
    private void setEmpate(int empate) {
        this.empate = empate;
    }
    
    //Metodos da Classe;
    public void apresentar(){
        System.out.println("====================================");
        System.out.println("Lutador: "+this.getNome()+".");
        System.out.println("Nacionalidade: "+this.getNacionalidade()+".");
        System.out.println("Idade: "+this.getIdade()+" anos.");
        System.out.println("Altura: "+this.getAltura()+" metros.");
        System.out.println("Peso: "+this.getPeso()+" Kg.");
        System.out.println("Categoria: "+this.getCategoria()+".");
        System.out.println("Vitorias: "+this.getVitorias()+".");
        System.out.println("Derrotas: "+this.getDerrotas()+".");
        System.out.println("Empates: "+this.getEmpate()+".");
    }
    public void status(){
        System.out.println("====================================");
        System.out.println("-------- "+this.getNome()+" --------");
        System.out.println("__ "+this.getPeso()+"Kg __");
        System.out.println("__ "+this.getCategoria()+" __");
        System.out.println("Vitorias: __ "+this.getVitorias()+" __");
        System.out.println("Derrotas: __ "+this.getDerrotas()+" __");
        System.out.println("Empates: __ "+this.getEmpate()+" __");
    }
    public void ganharLuta(){
        this.setVitorias(this.getVitorias() + 1);
    }
    public void perderLuta(){
        this.setDerrotas(this.getDerrotas() + 1);
    }
    public void empatarLuta(){
        this.setEmpate(this.getEmpate() + 1);
    }
    
}

File: POO PHP/Aula07/Lutador.php
<?php
class Lutador {

    //Atributos da Classe
    private $nome;
    private $nacionalidade;
    private $idade;
    private $altura;
    private $peso;
    private $categoria;
    private $vitoria;
    private $derrotas;
    private $empates;
    
    
    //Metodos Construtores
    public function __construct($nome, $nacionalidade, $idade, $altura, $peso, $vitoria, $derrotas, $empates) {
        $this->nome = $nome;
        $this->nacionalidade = $nacionalidade;
        $this->idade = $idade;
        $this->altura = $altura;
        $this->setPeso($peso);
        $this->vitoria = $vitoria;
        $this->derrotas = $derrotas;
        $this->empates = $empates;
    }

    
    //Metodos Acessores e Modificadores
    public function getNome() {
        return $this->nome;
    }
    public function getNacionalidade() {
        return $this->nacionalidade;
    }
    public function getIdade() {
        return $this->idade;
    }
    public function getAltura() {
        return $this->altura;
    }
    public function getPeso() {
        return $this->peso;
    }
    public function getCategoria() {
        return $this->categoria;
    }
    public function getVitoria() {
        return $this->vitoria;
    }
    public function getDerrotas() {
        return $this->derrotas;
    }
    public function getEmpates() {
        return $this->empates;
    }
    public function setNome($nome): void {
        $this->nome = $nome;
    }
    public function setNacionalidade($nacionalidade): void {
        $this->nacionalidade = $nacionalidade;
    }
    public function setIdade($idade): void {
        $this->idade = $idade;
    }
    public function setAltura($altura): void {
        $this->altura = $altura;
    }
    public function setPeso($peso): void {
        $this->peso = $peso;
        $this->setCategoria();
    }
    private function setCategoria(): void {
        if($this->peso < 52.2){
            $this->categoria = "Invalido";
        }elseif ($this->peso <= 70.3) {
            $this->categoria = "Leve";
        }elseif ($this->peso <= 83.9) {
            $this->categoria = "Medio";
        }elseif ($this->peso <= 120.2) {
            $this->categoria = "Pesado";
        } else {
            $this->categoria = "Invalido";
        }
    }
    public function setVitoria($vitoria): void {
        $this->vitoria = $vitoria;
    }
    public function setDerrotas($derrotas): void {
        $this->derrotas = $derrotas;
    }
    public function setEmpates($empates): void {
        $this->empates = $empates;
    }

        
    //Metodos da Classe
    public function apresentar(){
        echo "<p>-----------------------------------------</p>";
        echo "<p>Aqui apresento o lutador ".$this->getNome()." da ".$this->getNacionalidade();
        echo ".<br>Ele tem ".$this->getIdade()." anos.";
        echo " Pesa ".$this->getPeso()." Kg e altura de ".$this->getAltura()." metros.";
        echo " <br>Ele tem ".$this->getVitoria()." vitorias na carreira.";
        echo " Pedeu ".$this->getDerrotas()." lutas ";
        echo " <br>e por fim Empatou ".$this->getEmpates()." lutas.</p>";
    }
    public function status(){
        echo "<br><p>--------------------------------</p>";
        echo "<p>Nome: ".$this->getNome()."</p>";
        echo "--------------------------------";
        echo "<p>Tem Peso: ".$this->getCategoria()."</p>";
        echo "<p>Vitorias: ".$this->getVitoria()."</p>";
        echo "<p>Derrotas: ".$this->getDerrotas()."</p>";
        echo "<p>Empates: ".$this->getEmpates()."</p>";
    }
    public function ganharLuta(){
        $this->setVitoria($this->getVitoria() + 1);
    }
    public function perderLuta() {
        $this->setDerrotas($this->getDerrotas() + 1);
    }
    public function empatarLuta(){
        $this->setEmpates($this->getEmpates() + 1);
    }
}
